containers/python_function_deployment.md

```diff
@@ -0,0 +1,184 @@
+
+If you run into issues not listed here, please file an issue and we will attempt to find a generalizable approach to diagnosing and solving it.
+
+### Remote during container initialization
+**Some pip packages weren't installed**
+
+This is possible if conda is not tracking your pip packages. Make sure that you install your pip packages through a conda-installed `pip`. This way, conda can keep track of which pip packages you are using in your environment. You can see which pip packages conda is keeping track of by writing your environment to a file with `conda env export >> <filename>`
+
+**Logs indicate that some pip packages don't exist in the container's pip distribution library**
+
+If this occurs, `docker logs` should output something like the following during the 'Loading pip packages' phase:
+    
+  Collecting nb-conda-kernels==2.1.0
+    Could not find a version that satisfies the requirement nb-conda-kernels==2.1.0 (from versions: )
+  No matching distribution found for nb-conda-kernels==2.1.0
+
+  CondaValueError: Value error: pip returned an error.
+  
+This will not stop container startup; the pip package's installation will be skipped. If your Python function needs the package, you can run into runtime errors upon querying the container. The best way to avoid this issue is to ensure that all of your pip packages exist on linux-64 pip distribution library.
+
+### Errors at query time
+Runtime errors that occur within the container will crash it. You can diagnose what caused the issue by looking at the out put of `docker logs <container_id>` (where `<container_id>` is the id of the container in which your Python function was deployed)
+
+**Container crash: module not found**
+
+This can occur if:
+ 
+* Your function makes use of a pip package that your conda environment wasn't tracking. See the "Some pip packages weren't installed" section above.
+* Your function makes use of a conda package that doesn't exist in the container's conda channel and whose installation was skipped. If any packages in your local conda environment were skipped for installation on the container, you should have been notified of them in a warning message after calling `deploy_predict_function`.
+* Conda and pip both don’t have a dependency being used by the function. Unfortunately, Clipper's Python function deployment feature only supports conda and pip packages for now. Usage of modules not within these distributions is incompatible with this deployment process.
+
+**Container crash: prediction function has some other runtime error (index out of bounds, divide by zero, incorrect operand types for a function, etc**
+
+Make sure that you test your function locally before deploying it. Remember that it should take as input a list of datapoints (numpy arrays or native Python lists) with entries of type <input_type> and shoud output a list with a classification for each datapoint.

```

Comment:
typo s/shoud/should. Remind them once again that the output should be a list of strings.